service/api/client/client.go

```diff
@@ -60,30 +60,60 @@ func (c *Client) GetHostID(ctx context.Context) (peer.ID, error) {
 	return peer.IDFromBytes(resp.PeerID)
 }
 
-func (c *Client) CreateThread(ctx context.Context, id thread.ID) (info thread.Info, err error) {
-	resp, err := c.c.CreateThread(ctx, &pb.CreateThreadRequest{
+func (c *Client) CreateThread(ctx context.Context, id thread.ID, opts ...core.KeyOption) (info thread.Info, err error) {
+	args := &core.KeyOptions{}
+	for _, opt := range opts {
+		opt(args)
+	}
+	req := &pb.CreateThreadRequest{
 		ThreadID: id.Bytes(),
-	})
+		Keys:     &pb.ThreadKeys{},
+	}
+	if args.FollowKey != nil {
+		req.Keys.FollowKey = args.FollowKey.Bytes()
+	}
+	if args.ReadKey != nil {
+		req.Keys.ReadKey = args.ReadKey.Bytes()
+	}
+	if args.LogKey != nil {
+		req.Keys.LogKey, err = args.LogKey.Bytes()
+		if err != nil {
+			return
+		}
+	}
```

Comment:
^ All key combinations are possible. If you don't provide a follow key, you'll get errorswhen trying add records. In this sense, it's not "optional", but it's a tradeoff for a nicer API.

Reply:
If _no_ `LogKey`, then a private one is created internally. So, if you didn't supply a `ReadKey`, the remote would still not be able to write records. 

There's a matrix of possibilities here. I opted for all keys just being options w/ somewhat predictable consequences when omitted. Though, it's definitely not ideal. Open to alternatives here.

Reply:
Right, just trying to confirm if it was reasonable case or not. I'm OK with making this simple since combinations are a bunch to cover.